targets/TARGET_STM/pinmap.cpp

```diff
@@ -197,3 +198,27 @@ void pin_mode(PinName pin, PinMode mode)
     LL_HSEM_ReleaseLock(HSEM, CFG_HW_GPIO_SEMID, HSEM_CR_COREID_CURRENT);
 #endif /* DUAL_CORE */
 }
+
+/**
+ * List of peripherals excluded from testing
+ */
+const PeripheralList *pinmap_restricted_peripherals()
```

Comment:
Same here: I'd prefer not to impact the generic source code for tests purposed.
There is already a generic definition of the stdio_uart object that is used at platform level:
platform/source/mbed_retarget.cpp:extern serial_t stdio_uart;
Couldn't this be used for excluding stdio_uart from the test suite without changing the target code ? 

Moreover there is no other cpp files in the HAL, I feel like we're losing some consistency here

Reply:
I understand you don't like the `.c -> .cpp` change. The use of `pinmap_peripheral()` was my idea for covering a wider range of targets than those that define `STDIO_UART` peripheral.

I think that skipping the tests for https://github.com/ARMmbed/mbed-os/blob/236c3365263d15a93193e336eee5bd837f4561ee/platform/source/mbed_retarget.cpp#L148 is a good alternative. In general we use the `pinmap_restricted_peripherals()` for this purpose in FPGA tests.

@mprse will pick this PR next week. I'll let him decide how to update this.

Reply:
`serial_t stdio_uart` can not be used to initialize `peripherals[]` because it is not content.